File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  def self.compute(first, second)
    raise ArgumentError unless first.length == second.length
    self.loop_over_letters(first, second)
  end

  private
    def self.loop_over_letters(first, second)
      i = 0
      counter = 0
      while i != first.length do
        counter += 1 if first[i] != second[i]
        i += 1
      end
      return counter
    end
end

File: javascript/hamming/hamming.js
function Hamming() {

  this.compute = function ( sequence1, sequence2 ) {
    var counter = 0;

    if(sequence1.length !== sequence2.length) {
      throw new Error('DNA strands must be of equal length.');
    }

    for(var i = 0; i < sequence1.length; i++) {
        if(sequence1.charAt(i) !== sequence2.charAt(i)) {
            counter++;
        }
    }

    return counter;
  };
}

module.exports = Hamming;
